src/javascript/app/base/binary_loader.js

```diff
@@ -44,6 +44,7 @@ const BinaryLoader = (() => {
 
         container = getElementById('content-holder');
         container.addEventListener('binarypjax:before', beforeContentChange);
+        window.addEventListener('beforeunload', beforeContentChange);
```

Comment:
we don't need to remove the event later?

Reply:
I've didn’t find anywhere suitable for this but the 'beforeunload' itself. Do we need to remove it there?

Reply:
as long as loading different pages by clicking on menu items doesn't duplicate this event then it's fine, can you test it?

Reply:
Yes sure, I'll test.
Somehow I’m sure that it can’t be duplicated, because it’s on init function of loader. I'll inform you when I tested.

Reply:
Done, no duplication by clicking on menu items.